src/muse/readers/csv.py

```diff
@@ -403,6 +400,10 @@ def read_technologies(
     if "comm_type" in result.data_vars or "comm_type" in result.coords:
         result = result.drop_vars("comm_type")
 
+    check_utilization_and_minimum_service_factors(
```

Comment:
Just a note for if performance is a concern: Since this function is only ever used here, the internals of it could be adjusted to take an Xarray instead of a DataFrame

Reply:
It is not an issue because it's used only once and the size of the arrays is pretty small, so the overhead is minimal. I've given a try to adapt the code to use xarrays and it has become less readable - and didn't work (granted I only invested 15 min).

In summary, I'd leave it as it is.